src/frameworks/next/utils.ts

```diff
@@ -0,0 +1,67 @@
+import type { Header, Redirect, Rewrite } from "next/dist/lib/load-custom-routes";
+import type { Manifest, RoutesManifestRewrite } from "./interfaces";
+import { isUrl } from "../utils";
+
+/**
+ * Whether the given path has a regex or not.
+ * According to the Next.js documentation:
+ * ```md
+ *  To match a regex path you can wrap the regex in parentheses
+ *  after a parameter, for example /post/:slug(\\d{1,}) will match /post/123
+ *  but not /post/abc.
+ * ```
+ * See: https://nextjs.org/docs/api-reference/next.config.js/redirects#regex-path-matching
+ */
+export function pathHasRegex(path: string): boolean {
+  // finds parentheses that are not preceded by double backslashes
+  return /(?<!\\)\(/.test(path);
+}
+
+/**
+ * Remove double backslashes from a string
+ */
+export function cleanEscapedChars(path: string): string {
+  return path.replace(/\\/g, "");
+}
+
+/**
+ * Whether a Next.js rewrite is supported by Firebase.
+ */
+export function isRewriteSupportedByFirebase(rewrite: Rewrite): boolean {
+  if ("has" in rewrite || pathHasRegex(rewrite.source) || isUrl(rewrite.destination)) {
```

Comment:
What is the significance of "has" here? Can you add a comment explaining how this is being checked?

Reply:
@TheIronDev `has` is used for [Header, Cookie, and Query Matching](https://nextjs.org/docs/api-reference/next.config.js/rewrites#header-cookie-and-query-matching). As it can't be translated to firebase.json we say it's not supported. I improved the functions docs linking to both the `firebase.json` and Next.js docs